clickhouse/datadog_checks/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,292 @@
+            self.gauge('table.mergetree.part.current', part['parts'], tags=tags)
+            self.gauge('table.mergetree.row.current', part['rows'], tags=tags)
+
+    def query_system_replicas(self):
+        # https://clickhouse.yandex/docs/en/operations/system_tables/#system_tables-replicas
+        for replica in self.execute_query_defined_schema(queries.SystemReplicas):
+            tags = [
+                'database:{}'.format(replica['database']),
+                'table:{}'.format(replica['table']),
+                'is_leader:{}'.format(boolean_string(replica['is_leader'])),
+                'is_readonly:{}'.format(boolean_string(replica['is_readonly'])),
+                'is_session_expired:{}'.format(boolean_string(replica['is_session_expired'])),
+            ]
+            tags.extend(self._tags)
+
+            self.gauge('table.replicated.part.future', replica['future_parts'], tags=tags)
+            self.gauge('table.replicated.part.suspect', replica['parts_to_check'], tags=tags)
+            self.gauge('table.replicated.version', replica['columns_version'], tags=tags)
+            self.gauge('table.replicated.queue.size', replica['queue_size'], tags=tags)
+            self.gauge('table.replicated.queue.insert', replica['inserts_in_queue'], tags=tags)
+            self.gauge('table.replicated.queue.merge', replica['merges_in_queue'], tags=tags)
+            self.gauge('table.replicated.log.max', replica['log_max_index'], tags=tags)
+            self.gauge('table.replicated.log.pointer', replica['log_pointer'], tags=tags)
+            self.gauge('table.replicated.total', replica['total_replicas'], tags=tags)
+            self.gauge('table.replicated.active', replica['active_replicas'], tags=tags)
+
+    def query_system_dictionaries(self):
+        # https://clickhouse.yandex/docs/en/operations/system_tables/#system-dictionaries
+        for dictionary in self.execute_query_defined_schema(queries.SystemDictionaries):
+            tags = ['dictionary:{}'.format(dictionary['name'])]
+            tags.extend(self._tags)
+
+            self.gauge('dictionary.memory.used', dictionary['bytes_allocated'], tags=tags)
+            self.gauge('dictionary.item.current', dictionary['element_count'], tags=tags)
+            self.gauge('dictionary.load', dictionary['load_factor'], tags=tags)
+
+    def query_custom(self):
+        for custom_query in self._custom_queries:
+            query = custom_query.get('query')
+            if not query:  # no cov
+                self.log.error('Custom query field `query` is required')
+                continue
+
+            columns = custom_query.get('columns')
+            if not columns:  # no cov
+                self.log.error('Custom query field `columns` is required')
+                continue
+
+            self.log.debug('Running custom query for ClickHouse')
+            try:
+                rows = self.execute_query_raw(query, 'custom query', stream=True)
```

Comment:
AFAIU it will fail when it fails (?!) because we expect the second argument to be a callable.